corehq/ex-submodules/couchforms/util.py

```diff
@@ -672,6 +672,10 @@ def _handle_unexpected_error(e, instance):
     instance.__class__ = XFormError
     instance.orig_id = instance._id
     instance._id = new_id
+    if '_rev' in instance:
+        # clear the rev since we want to make a new doc
+        # this is necessary for errors that come from editing submissions
+        del instance['_rev']
```

Comment:
i haven't seen the 500s on prod so I think maybe it only triggers in the edit workflow. i didn't investigate close enough to know whether it happens under other circumstances